plugins/modules/win_domain_computer.ps1

```diff
@@ -146,6 +157,73 @@ Function Add-ConstructedState($desired_state) {
   $result.changed = $true
 }
 
+Function Invoke-OfflineDomainJoin {
+  [CmdletBinding(SupportsShouldProcess=$true)]
+  param(
+    [Parameter(Mandatory=$true)]
+    [System.Collections.IDictionary]
+    $desired_state ,
+
+    [Parameter()]
+    [bool]
+    $ReturnBlob = $false ,
+
+    [Parameter()]
+    [System.IO.FileInfo]
+    $BlobPath
+  )
+
+  End {
+    $dns_domain = $desired_state.dns_hostname -replace '^[^.]+\.'
+
+    $delete = -not $BlobPath
+    if ($delete) {
+      $BlobPath = [System.IO.Path]::GetTempFileName()
```

Comment:
While it's not as easy to access in the older style modules you can access the temp path set by Ansible with

```powershell
$temp = Get-AnsibleParam -obj $params -name '_ansible_remote_tmp' -type 'path' -default $env:TEMP
```

From there you can do `$BlobPath = Join-Path -Path $temp -ChildPath ([IO.Path]::GetRandomFileName())`. The only other thing I suggest is to wrap this in a try/catch to ensure `$BlobPath` is deleted if it exists and we don't need to keep it